pynumdiff/linear_model/_linear_model.py

```diff
@@ -76,24 +76,27 @@ def polydiff(x, dt, params=None, options=None, polynomial_order=None, window_siz
              - **dxdt_hat** -- estimated derivative of x
     """
     if params != None:
-        warn("`params` and `options` parameters will be removed in a future version. Use `polynomial_order` " +
+        warn("`params` and `options` parameters will be removed in a future version. Use `poly_order` " +
             "and `window_size` instead.", DeprecationWarning)
-        polynomial_order = params[0]
+        poly_order = params[0]
         if len(params) > 1: window_size = params[1]
         if options != None:
             if 'sliding' in options and not options['sliding']: window_size = None
             if 'step_size' in options: step_size = options['step_size']
             if 'kernel_name' in options: kernel = options['kernel_name']
-    elif polynomial_order == None or window_size == None:
-        raise ValueError("`polynomial_order` and `window_size` must be given.")
+    elif poly_order == None or window_size == None:
+        raise ValueError("`poly_order` and `window_size` must be given.")
 
-    if window_size < polynomial_order*3:
-        window_size = polynomial_order*3+1
+    if window_size < poly_order*3:
+        window_size = poly_order*3+1
+    if window_size % 2 == 0:
+        window_size += 1
```

Comment:
This thing was failing because the optimizer wanted to give even-length windows, so I've added back this `+1`.